### ISSUE TITLE
How do I install companion expansion with latest release? 

### ISSUE THREAD

**Issue by apollodeathstar** [phobos2077/fo2_ecco#14]
Can I use the patch from a few releases ago?

**Comment by phobos2077** [comment]
No, it will revert certain changes in the mod. Someone needs to make a new patch and I didn't want to do it because I don't play with this mod myself.

**Comment by apollodeathstar** [comment]
> No, it will revert certain changes in the mod. Someone needs to make a new patch and I didn't want to do it because I don't play with this mod myself.

how did you make the last patch? what would I have to do?

**Comment by phobos2077** [comment]
> how did you make the last patch? what would I have to do?

You need to use dat2 or Dat Explorer to unpack dat files of both mods, then use compare tool like Beyond Compare or Win Merge to find conflicting files and merge changes. Then you'd also need to merge conflicting scripts (int) files, which is a bit more complicated... 

Maybe I'll do it myself at a later date when both mods are more stable and won't be updating so often.

**Comment by apollodeathstar** [comment]
> > how did you make the last patch? what would I have to do?
> 
> You need to use dat2 or Dat Explorer to unpack dat files of both mods, then use compare tool like Beyond Compare or Win Merge to find conflicting files and merge changes. Then you'd also need to merge conflicting scripts (int) files, which is a bit more complicated...
> 
> Maybe I'll do it myself at a later date when both mods are more stable and won't be updating so often.

Sounds good hope you get around to it eventually. I will give it a shot with I have some time this week.

**Comment by Lichtenshtein** [comment]
I found only one conflicting ai.txt. Maybe the mods are compatible with the latest release of Companion Expansion? I merged both ai.txt just in case and added them as a patch for my mods_order.txt .

**Comment by phobos2077** [comment]
> I found only one conflicting ai.txt. Maybe the mods are compatible with the latest release of Companion Expansion? I merged both ai.txt just in case and added them as a patch for my mods_order.txt .

I'm pretty sure they aren't compatible without manually merging the changes, as you did. Because EcCo changes existing AI packets while CE adds new ones.

**Comment by Lichtenshtein** [comment]
> I'm pretty sure they aren't compatible without manually merging the changes, as you did. Because EcCo changes existing AI packets while CE adds new ones.

Well, yes i saw EcCo changes some freq numbers but i didn't think they are critical to mods compability. If so then maybe you are right and manual merging is required.

I think companion expansion does conflict with latest RPU...for example, there was no Miria on a map at all in Modoc, i just didn't meet her in my playthrough. Also some talking heads of the standard companions were mixed with the heads of some NPCs, like after Sulik joined, he got the head of Lara, and then Angela Bishop's. And Cassidy became that black Jo dude from Modoc :D Nevertheless, the mod is playable with RPU and other mods, these old new characters are cool...
